src/rapids_dependency_file_generator/rapids_dependency_file_generator.py

```diff
@@ -188,8 +161,33 @@ def make_dependency_file(
                     "'table' is not one of 'build-system' or 'project'."
                 )
 
+        # This file type needs to be modified in place instead of built from scratch.
```

Comment:
Did you move all the code just so we can error earlier in the above cases? AFAICT nothing actually changed here other than that you call tomlkit.dumps on the end result, which I guess is to isolate the divergence in file types here from where it used to be in `make_dependency_files`.

Reply:
> Did you move all the code just so we can error earlier in the above cases?

Yes.

> you call tomlkit.dumps on the end result, which I guess is to isolate the divergence in file types here from where it used to be in `make_dependency_files`.

Correct.